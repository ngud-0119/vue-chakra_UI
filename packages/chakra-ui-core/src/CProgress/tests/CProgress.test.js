import { CProgress, CProgressLabel } from '../..'
import { render } from '@/tests/test-utils'

const renderComponent = (props) => {
  const base = {
    components: {
      CProgress,
      CProgressLabel
    },
    template: '<CProgress rounded="sm" color="green" size="sm" data-testid="progress" :value="40" />',
    ...props
  }
  return render(base)
}

it('should render correctly', () => {
  const { asFragment } = renderComponent()
  expect(asFragment()).toMatchSnapshot()
})

it('should have the correct width', () => {
  const { queryByRole } = renderComponent()

  expect(queryByRole('progressbar')).toHaveStyle('width: 40%')
})

it('should display a label', () => {
  const { queryByText } = renderComponent({
    template: `
  <CProgress :value="40">
    <CProgressLabel>Label</CProgressLabel>
  </CProgress>`
  })

  expect(queryByText('Label')).toBeInTheDocument()
})

test('a11y - progress has a "role" set to "progressbar"', () => {
  const { queryByRole } = renderComponent()
  expect(queryByRole('progressbar')).toBeInTheDocument()
})

test('a11y - progress has a "aria-valuenow" set to the percentage completion value', () => {
  const { queryByRole } = renderComponent()

  expect(queryByRole('progressbar')).toHaveAttribute(
    'aria-valuenow',
    '40'
  )
})
