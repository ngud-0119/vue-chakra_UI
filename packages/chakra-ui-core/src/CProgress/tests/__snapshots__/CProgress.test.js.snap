// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  position: relative;
  height: 0.5rem;
  overflow: hidden;
  width: 100%;
  background: var(--chakra-colors-gray-100);
  border-radius: var(--chakra-radii-sm);
}

.emotion-1 {
  height: 100%;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  width: 40%;
  background-color: var(--chakra-colors-green-500);
  border-radius: var(--chakra-radii-sm);
}

<div
    class="emotion-0"
    data-chakra-component="CProgress"
    data-testid="progress"
  >
    <div
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="40"
      class="emotion-1"
      data-chakra-component="CProgressIndicator"
      role="progressbar"
    />
  </div>
</DocumentFragment>
`;
